browser/src/slideshow/PauseTimer.ts

```diff
@@ -81,108 +76,13 @@ class PauseTimerGl extends Transition2d implements PauseTimer {
 		}
 	}
 
-	private createTextTexture(remainingCount: number): WebGLTexture {
-		const displayText = `Pause...( ${remainingCount} )`;
+	public createTextTexture(displayText: string): WebGLTexture {
 		const textCanvas = this.create2DCanvasWithText(displayText);
 		return this.load2dCanvasToGlCanvas(textCanvas);
 	}
 
-	// Create an off-screen 2D canvas with text centered
-	private create2DCanvasWithText(displayText: string): HTMLCanvasElement {
-		const canvas = document.createElement('canvas');
-		canvas.width = this.context.canvas.width;
-		canvas.height = this.context.canvas.height;
-		const ctx = canvas.getContext('2d');
-
-		// Set background color
-		ctx.fillStyle = 'black';
-		ctx.fillRect(0, 0, canvas.width, canvas.height);
-
-		// Set text attributes
-		ctx.fillStyle = 'white';
-		ctx.font = '20px sans-serif';
-		ctx.textAlign = 'center';
-		ctx.textBaseline = 'middle';
-
-		ctx.fillText(displayText, canvas.width / 2, canvas.height / 2);
-
-		return canvas;
-	}
-
-	// TODO: We can replace with loadTexture from RenderContext
-	private load2dCanvasToGlCanvas(canvas: HTMLCanvasElement): WebGLTexture {
-		const texture = this.gl.createTexture();
-		if (!texture) {
-			throw new Error('Failed to create texture');
-		}
-		this.gl.bindTexture(this.gl.TEXTURE_2D, texture);
-		this.gl.texImage2D(
-			this.gl.TEXTURE_2D,
-			0,
-			this.gl.RGBA,
-			this.gl.RGBA,
-			this.gl.UNSIGNED_BYTE,
-			canvas,
-		);
-		this.gl.texParameteri(
-			this.gl.TEXTURE_2D,
-			this.gl.TEXTURE_WRAP_S,
-			this.gl.CLAMP_TO_EDGE,
-		);
-		this.gl.texParameteri(
-			this.gl.TEXTURE_2D,
-			this.gl.TEXTURE_WRAP_T,
-			this.gl.CLAMP_TO_EDGE,
-		);
-		this.gl.texParameteri(
-			this.gl.TEXTURE_2D,
-			this.gl.TEXTURE_MIN_FILTER,
-			this.gl.LINEAR,
-		);
-		this.gl.texParameteri(
-			this.gl.TEXTURE_2D,
-			this.gl.TEXTURE_MAG_FILTER,
-			this.gl.LINEAR,
-		);
-
-		return texture;
-	}
-
-	public render(): void {
-		this.gl.viewport(
-			0,
-			0,
-			this.context.canvas.width,
-			this.context.canvas.height,
-		);
-		this.gl.clearColor(0.0, 0.0, 0.0, 1.0);
-		this.gl.clear(this.gl.COLOR_BUFFER_BIT);
-
-		this.gl.useProgram(this.program);
-
-		this.gl.activeTexture(this.gl.TEXTURE0);
-		this.gl.bindTexture(this.gl.TEXTURE_2D, this.textTexture);
-
-		const textureLocation = this.gl.getUniformLocation(
-			this.program,
-			'u_texture',
-		);
-		this.gl.uniform1i(textureLocation, 0);
-
-		this.gl.bindVertexArray(this.vao);
-		this.gl.drawArrays(this.gl.TRIANGLE_STRIP, 0, 4);
-	}
-
-	public getFragmentShader(): string {
-		return `#version 300 es
-			precision highp float;
-			in vec2 v_texCoord;
-			uniform sampler2D u_texture;
-			out vec4 outColor;
-
-			void main() {
-				outColor = texture(u_texture, v_texCoord);
-			}`;
+	private getPauseTextContent(): string {
+		return _(`Pause...( ${Math.ceil(this.pauseTimeRemaining)} )`);
```

Comment:
this should be `_('Pause...') + time` or `_('Pause... ( %SECONDS% )').replace('%SECONDS', value);` because it will not work with dynamically changed number :)

Reply:
updated.